exist-core/src/main/java/org/exist/xquery/ExternalModuleImpl.java

```diff
@@ -242,8 +242,7 @@ public Variable resolveVariable(QName qname) throws XPathException {
 
     public void analyzeGlobalVars() throws XPathException {
         for (final VariableDeclaration decl : mGlobalVariables.values()) {
-            decl.resetState(false);
-            decl.analyze(new AnalyzeContextInfo());
+            decl.analyzeExpression(new AnalyzeContextInfo());
```

Comment:
Just to check - is it okay to not call `decl.ResetState(false)`?

Reply:
Yes, we're calling this method right after the module was compiled, so there's no state to reset yet.